airflow/jobs/scheduler_job_runner.py

```diff
@@ -1486,7 +1465,12 @@ def _update_state(dag: DAG, dag_run: DagRun):
 
             dag_run.state = DagRunState.RUNNING
             dag_run.start_date = timezone.utcnow()
-            if dag.timetable.periodic and not dag_run.external_trigger and dag_run.clear_number < 1:
+            if (
+                dag.timetable.periodic
+                and dag_run.triggered_by != DagRunTriggeredByType.ASSET
+                and not dag_run.external_trigger
+                and dag_run.clear_number < 1
+            ):
```

Comment:
Why do we need to change this?

Reply:
because if we set schedule to something like `AssetOrTimeSchedule(timetable=CronTriggerTimetable("0 1 * * 3", timezone="UTC"), assets=dag1_asset_2)`. the timetable will be periodic even if it's triggered by asset which causes the error as there are no data interval for such dag runs. Also according to the comment 

```
                # We only publish these metrics for scheduled dag runs and only
                # when ``external_trigger`` is *False* and ``clear_number`` is 0.
```

asset triggered dag runs should not be included

Reply:
Hmm, OK. We also can’t get rid of `periodic` because e.g. `@once` would have a scheduled run but that shouldn’t be counted as periodic.